DigitalLearningSolutions.Web/Controllers/FrameworksController/Signposting.cs

```diff
@@ -1,22 +1,79 @@
 ﻿using DigitalLearningSolutions.Web.ViewModels.Frameworks;
 using Microsoft.AspNetCore.Mvc;
+using System;
+using System.IO;
 using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using DigitalLearningSolutions.Data.ApiClients;
+using System.Net.Http;
+using Microsoft.Extensions.Configuration;
 
 namespace DigitalLearningSolutions.Web.Controllers.FrameworksController
 {
     public partial class FrameworksController
     {
+        private static IConfiguration SignpostingConfiguration { get; set; }
+
         [Route("/Frameworks/{frameworkId}/Competency/{frameworkCompetencyId}/CompetencyGroup/{frameworkCompetencyGroupId}/Signposting")]
-        public IActionResult EditCompetencyLearningResources(int frameworkId, int? frameworkCompetencyGroupId = null, int? frameworkCompetencyId = null)
+        public IActionResult EditCompetencyLearningResources(int frameworkId, int? frameworkCompetencyGroupId = null, int? frameworkCompetencyId = null, string title = "")
         {
             var model = PopulatedModel(frameworkId, frameworkCompetencyId, frameworkCompetencyId);
+            model.Title = title;
             return View("Developer/EditCompetencyLearningResources", model);
         }
 
         [Route("/Frameworks/{frameworkId}/Competency/{frameworkCompetencyId}/CompetencyGroup/{frameworkCompetencyGroupId}/AddResources")]
-        public IActionResult AddCompetencyLearningResources(int frameworkId, int? frameworkCompetencyGroupId = null, int? frameworkCompetencyId = null)
+        public IActionResult AddCompetencyLearningResources(int frameworkId, int? frameworkCompetencyGroupId = null, int? frameworkCompetencyId = null, string Title = "")
         {
             var model = PopulatedModel(frameworkId, frameworkCompetencyId, frameworkCompetencyId);
+            model.Title = Title;
+            return View("Developer/AddCompetencyLearningResources", model);
+        }
+
+        [HttpPost]
```

Comment:
When I hit this page, the ViewModel is is populated as URL query parameters:
`https://localhost:44363/Frameworks/AddCompetencyLearningResourceSummary?referenceId=298&frameworkId=2&frameworkCompetencyId=1&frameworkCompetencyGroupId=1&resourceName=I%20can%20find,%20use%20and%20store%20information%20that%20exists%20in%20different%20digital%20locations&resourceType=Article&catalogue=Community%20contributions&description=%3Cp%3ETest%3C%2Fp%3E%0A&nameOfCompetency=I%20can%20organise%20my%20information%20and%20content%20using%20files%20and%20folders%20(either%20on%20my%20device,%20across%20multiple%20devices,%20or%20on%20the%20Cloud)&searchText=test`
I would prefer for us to using a HttpGet controller method to load the data into the ViewModel for the summary page rather than passing it using route values on the calling button in _SignpostingResourceCard.cshtml. This will make the page and method independently testable.

Reply:
Also, if I refresh this page I get a 500 error which I should never get.